File: library/src/main/java/com/yzyfdf/library/base/BaseApplication.java
package com.yzyfdf.library.base;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;

import androidx.multidex.MultiDex;

import com.billy.android.loading.Gloading;
import com.blankj.utilcode.util.Utils;
import com.yzyfdf.library.view.GlobalLoadingAdapter;

/**
 * APPLICATION
 */
public class BaseApplication extends Application {

    private static BaseApplication baseApplication;
    private static int             statusBarHeight;

    @Override
    public void onCreate() {
        super.onCreate();
        baseApplication = this;
        Utils.init(this);
        Gloading.initDefault(new GlobalLoadingAdapter());
    }

    public static Context getAppContext() {
        return baseApplication;
    }

    public static Resources getAppResources() {
        return baseApplication.getResources();
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
    }

    /**
     * 分包
     *
     * @param base
     */
    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

}

File: library/src/main/java/com/yzyfdf/library/base/BaseApplication.kt
package com.yzyfdf.library.base

import android.app.Application
import android.content.Context
import android.content.res.Resources
import androidx.multidex.MultiDex
import com.blankj.utilcode.util.Utils
import com.yzyfdf.library.view.globalloading.Gloading
import com.yzyfdf.library.view.globalloading.GlobalLoadingAdapter

/**
 * APPLICATION
 */
open class BaseApplication : Application() {

    companion object {

        private lateinit var baseApplication: BaseApplication

        val appContext: Context
            get() = baseApplication

        val appResources: Resources
            get() = baseApplication.resources
    }


    override fun onCreate() {
        super.onCreate()
        baseApplication = this
        Utils.init(this)
        Gloading.initDefault(GlobalLoadingAdapter())
        Gloading.debug(true)
    }

    override fun onTerminate() {
        super.onTerminate()
    }

    /**
     * 分包
     *
     * @param base
     */
    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

}
